pkg/yang/types_builtin.go

```diff
@@ -588,17 +699,42 @@ func (n Number) Less(m Number) bool {
 		return true
 	case n.Kind != Negative && m.Kind == Negative:
 		return false
-	case n.Kind == Negative:
-		return n.Value > m.Value
-	default:
-		return n.Value < m.Value
 	}
+
+	nt, mt := n.Trunc(), m.Trunc()
+	nf, mf := n.Frac(), m.Frac()
```

Comment:
Might as well move this inside the the nt == mt block since we only need it then.